src/types/base.ts

```diff
@@ -23,26 +23,37 @@ type MyInfoSourceDefault = Exclude<MyInfoSource, MyInfoSource.NotApplicable>
 // For a full reference, see https://www.ndi-api.gov.sg/library/myinfo/implementation-myinfo-data
 export type MyInfoNotApplicable = SourceProp<MyInfoSource.NotApplicable>
 
-export type MyInfoApplicable<S> = {
-  lastupdated: string
+export type MyInfoApplicable<S extends MyInfoSource> = {
+  // The “lastupdated” field is not always non-empty currently, but under what circumstances this occurs is unknown
+  lastupdated?: string
   classification: MyInfoDataClassification.Confidential
 } & SourceProp<S>
 
 type UnavailableProp<T> = {
   unavailable: T
 }
 
-export type MyInfoUnavailableField<S> = MyInfoApplicable<S> &
-  UnavailableProp<true>
-
-export type MyInfoAvailableMetadata<S> = MyInfoApplicable<S> &
-  Partial<UnavailableProp<undefined>> // For convenience
-
-export type MyInfoField<T, S = MyInfoSourceDefault> =
+export type MyInfoUnavailableField<
+  S extends MyInfoSource
+> = MyInfoApplicable<S> & UnavailableProp<true>
+
+export type MyInfoAvailableMetadata<
+  S extends MyInfoSource
+> = MyInfoApplicable<S> & Partial<UnavailableProp<undefined>>
+// Note on Partial<UnavailableProp<undefined>>:
+//
+// This is included for syntactic convenience so that code such as the below can be compiled in TypeScript:
+//
+//   if (!data.unavailable) {...}
+//
+// In reality, if it exists, the value of "unavailable" property is always false.
+// This is a design quirk of the MyInfo API.
```

Comment:
always `false` or always `true`? 🤔

Reply:
The value is always false, if the property exists

Reply:
But... what about this:
```
export type MyInfoUnavailableField<
  S extends MyInfoSource
> = MyInfoApplicable<S> & UnavailableProp<true>
```

Reply:
You are right, that was an accident